base/setup/usetup/chkdsk.c

```diff
@@ -56,11 +56,13 @@ ChkdskCallback(
 NTSTATUS
 ChkdskPartition(
     IN PUNICODE_STRING DriveRoot,
-    IN PFILE_SYSTEM_ITEM FileSystem)
+    /*IN PFILE_SYSTEM_ITEM FileSystemItem*/
+    IN PFILE_SYSTEM FileSystem)
 {
     NTSTATUS Status;
+    // PFILE_SYSTEM FileSystem = FileSystemItem->FileSystem;
 
-    if (!FileSystem->ChkdskFunc)
+    if (!FileSystem || !FileSystem->ChkdskFunc)
```

Comment:
Is calling this with a NULL FileSystem legal?
If not, could assert on it.

Reply:
I prefer keeping it like this, and I've adapted the calling code to check for the STATUS_NOT_SUPPORTED code and take appropriate action.